File: Programming Basics with Python/T06 - While Loop/exercise/04. Walking.py
total_steps = 0
target_steps = 10000
command = input()

while target_steps > total_steps:
    if command == "Going home":
        command = input()
        steps = int(command)
        total_steps += steps
        if total_steps >= target_steps:
            print("Goal reached! Good job!")
            print(f"{abs(target_steps - total_steps)} steps over the goal!")

        else:
            print(f"{abs(target_steps - total_steps)} more steps to reach goal.")
            break

    else:
        steps = int(command)
        total_steps += steps
        if total_steps >= target_steps:
            print("Goal reached! Good job!")
            print(f"{abs(total_steps - target_steps)} steps over the goal!")
            break
        command = input()

File: Programming Basics with Go/T06 - Advanced Loops/exercise/04. Walking.go
package main

import (
	"fmt"
	"strconv"
)

func main() {

	//входни данни (текст): "GoingHome" или извървяни стъпки ("123")
	//спираме: входни данни == "GoingHome"
	//продължаваме: входни данни != "GoingHome"

	var targetSteps int = 10000  //целеви стъпки
	var totalWalkedSteps int = 0 //общо извървяни стъпки

	var input string
	fmt.Scanln(&input)

	for input != "GoingHome" {
		//извървяни стъпки ("567")
		//input = "445"
		walkedSteps, _ := strconv.Atoi(input) //текущи извървяни стъпки

		//добавяме към общия бр извървяни стъпки
		totalWalkedSteps += walkedSteps

		//проверка дали стигаме целта
		if totalWalkedSteps >= targetSteps {
			fmt.Println("Goal reached! Good job!")
			fmt.Printf("%d steps over the goal!", totalWalkedSteps-targetSteps)
			break
		}

		fmt.Scanln(&input)
	}

	if input == "GoingHome" {
		var stepsToHome int
		fmt.Scanln(&stepsToHome)
		totalWalkedSteps += stepsToHome
		//след прибирането е успяла да стигне 10000
		if totalWalkedSteps >= targetSteps {
			fmt.Println("Goal reached! Good job!")
			fmt.Printf("%d steps over the goal!", totalWalkedSteps-targetSteps)
		} else {
			//не е успяла -> totalWalkedSteps < targetSteps
			fmt.Printf("%d more steps to reach goal.", targetSteps-totalWalkedSteps)
		}

	}
}
